scipy/sparse/tests/test_construct.py

```diff
@@ -471,3 +471,9 @@ def test_random_accept_str_dtype(self):
         # for the dtype
         construct.random(10, 10, dtype='d')
 
+    def test_random_sparse_matrix_returns_correct_number_of_non_zero_elements(self):
+        # A 10 x 10 matrix, with density of 12.65%, should have 13 nonzero elements.
+        # 10 x 10 x 0.1265 = 12.65, which should be rounded up to 13, not 12.
```

Comment:
This may be a little misleading. You're trying to solve rounding errors only right (e.g. `10.999...` instead of `11`)? It's not possible to get `12.65` here ever, or if it is there's a different bug somewhere.